### PR TITLE
Feat: Remove proxy

### PR DESCRIPTION
**Pull request by egor0798**
## Changes:
- Added remove proxy flow UI
- Updated `Signing` to perform signing with not active wallet

***

image

### CODE REVIEW

**Comment by tuul-wq** [comment]
Task linked: [CU-86933z3nv Remove proxy account](https://app.clickup.com/t/86933z3nv)

File: src/renderer/entities/proxy/model/proxy-model.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import { combine, createEffect, createEvent, createStore, sample } from 'effector';
-import groupBy from 'lodash/groupBy';
+import { groupBy } from 'lodash';
```
Comment (Asmadek) [inline src/renderer/entities/proxy/model/proxy-model.ts:2]: What is the reason for this changes? Maybe better to use [this approach](https://tyapk.ru/blog/post/correct-way-to-import-lodash-methods) as we do in the most of places

File: src/renderer/entities/transaction/lib/extrinsicService.ts [hunk 2]
```diff
@@ -302,6 +302,13 @@ export const getUnsignedTransaction: Record<
       options,
     );
   },
+  [TransactionType.REMOVE_PROXY]: (transaction, info, options) => {
+    return methods.proxy.removeProxy(
+      { delegate: transaction.args.delegate, proxyType: transaction.args.proxyType, delay: transaction.args.delay },
```
Comment (Asmadek) [inline src/renderer/entities/transaction/lib/extrinsicService.ts:307]: Looks like better to divide it for lines

File: src/renderer/entities/transaction/model/transaction.ts [hunk 3]
```diff
@@ -24,6 +24,8 @@ export const enum TransactionType {
   BATCH_ALL = 'batchAll',
   DESTINATION = 'payee',
   CHILL = 'chill',
+
+  REMOVE_PROXY = 'remove_pryx',
```
Comment (Asmadek) [inline src/renderer/entities/transaction/model/transaction.ts:28]: is it typo?

File: src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+  CONFIRMATION,
+  SIGNING,
+  SUBMIT,
+}
+
+type FlowStartedProps = {
+  chain: Chain;
+  proxyAccount: ProxyAccount;
+};
+const flowStarted = createEvent<FlowStartedProps>();
```
Comment (Asmadek) [inline src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts:22]: what do you think about more meaningful `removeStarted`?

File: src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+    return (
+      accounts.find(
+        (a) => a.accountId === proxyAccount.proxiedAccountId && !walletUtils.isWatchOnly(walletsMap[a.walletId]),
+      ) || null
+    );
+  },
+  {
+    skipVoid: false,
+  },
```
Comment (Asmadek) [inline src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts:51]: better to use one line here

File: src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+    proxiedAccount: $proxiedAccount,
+  },
+  ({ wallets, proxiedAccount }) => {
+    if (!proxiedAccount) return null;
+
+    return wallets.find((w) => w.id === proxiedAccount.walletId) || null;
+  },
+  {
+    skipVoid: false,
+  },
```
Comment (Asmadek) [inline src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts:66]: better to use one line here

File: src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+  ({ wallets, accounts, proxiedAccount, chain }) => {
+    if (!chain || !proxiedAccount) return null;
+
+    return accountUtils.isMultisigAccount(proxiedAccount)
+      ? getSignatoryAccounts(accounts, wallets, [], proxiedAccount.signatories, chain.chainId)
+      : null;
+  },
+  {
+    skipVoid: false,
+  },
```
Comment (Asmadek) [inline src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts:85]: better to use one line here

File: src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+);
+
+const signatorySelected = createEvent<Account>();
```
Comment (Asmadek) [inline src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts:88]: Better to group events before stores

File: src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+    );
+
+    if (sameGroupProxies.length === 1 && proxyGroup) {
+      return { proxyGroup, shouldDelete: true };
+    }
+
+    const updatedDeposit =
+      Number(proxyGroup!.totalDeposit) -
+      api.consts.proxy.proxyDepositBase.toNumber() -
+      api.consts.proxy.proxyDepositBase.toNumber();
```
Comment (Asmadek) [inline src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts:160]: Better to use `BN` here

And maybe you should subtract `proxyDepositFactor` not `proxyDepositBase`? As I clearly understand `proxyDepositBase` used only for first (last on remove) proxy

File: src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+sample({
+  clock: proxyRemoved,
+  source: {
+    proxyAccount: $proxyAccount,
+    proxies: proxyModel.$proxies,
+    proxyGroups: proxyModel.$proxyGroups,
+  },
+  filter: (proxyAccount) => Boolean(proxyAccount),
+  fn: ({ proxyAccount, proxies, proxyGroups }, api) => {
+    const sameGroupProxies = proxies[proxyAccount!.proxiedAccountId].filter((p) => p.chainId === proxyAccount!.chainId);
```
Comment (Asmadek) [inline src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts:148]: What is this variable for? maybe use another name or add comment to describe logic here?

**Review by Asmadek — CHANGES_REQUESTED** [review]

File: src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts (hunk 8 above)
Comment (tuul-wq) [reply src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts:88]: Agree, link to our [tg](https://t.me/c/1926452604/963) discussion regarding grouping of events and stores

File: src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts (hunk 4 above)
Comment (tuul-wq) [reply src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts:22]: Agree, we control the naming, no need for unnecessary abstraction

File: src/renderer/entities/proxy/model/proxy-model.ts (hunk 1 above)
Comment (egor0798) [reply src/renderer/entities/proxy/model/proxy-model.ts:2]: I think I had merge conflicts with your request so I was changing `dictionery` back to `groupBy`and webstorm imported it like that
changed back to how it was.

File: src/renderer/entities/transaction/model/transaction.ts (hunk 3 above)
Comment (egor0798) [reply src/renderer/entities/transaction/model/transaction.ts:28]: yep

File: src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts (hunk 9 above)
Comment (egor0798) [reply src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts:160]: yeah, second one should be `proxyDepositFactor`

File: src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts (hunk 10 above)
Comment (egor0798) [reply src/renderer/widgets/RemoveProxy/model/remove-proxy-model.ts:148]: added comment

**Review by tuul-wq — CHANGES_REQUESTED** [review]

File: src/renderer/shared/assets/images/walletTypes/proxy.svg
[the remark this answers is not in the document]
Comment (egor0798) [reply src/renderer/shared/assets/images/walletTypes/proxy.svg:1]: Last one was old and different from design system one. Changed to `42x42` but left the viewbox the same otherwise it looks misaligned

File: src/renderer/widgets/WalletDetails/ui/components/ProxiesList.tsx [hunk 11]
```diff
@@ -33,15 +33,15 @@ export const ProxiesList = ({ walletId, className, canCreateProxy = true }: Prop
         <FootnoteText className="flex-1 px-2 text-text-tertiary">{t('accountList.addressColumn')}</FootnoteText>
       </div>
 
-      <ul className="flex flex-col h-full px-5 divide-y divide-divider overflow-y-auto overflow-x-hidden">
+      <ul className="flex flex-col px-5 divide-y divide-divider overflow-y-auto overflow-x-hidden h-[inherit]">
```
[the remark this answers is not in the document]
Comment (egor0798) [reply src/renderer/widgets/WalletDetails/ui/components/ProxiesList.tsx:36]: Changed it back. Wiht only one proxy in the list it was centered in the modal but since I removed `flex-1` from proxy conponent below I reverted to 'h-full' now

File: src/renderer/features/proxies/lib/proxies-utils.ts [hunk 12]
```diff
@@ -39,8 +39,8 @@ function getNotification({
       dateCreated: Date.now(),
       proxyType: proxied.proxyType,
       proxyAccountId: proxied.proxyAccountId,
-      proxyWalletName: accountsWalletsMap[proxied.proxyAccountId].name,
-      proxyWalletType: accountsWalletsMap[proxied.proxyAccountId].type,
+      proxyWalletName: accountsWalletsMap[proxied.proxyAccountId]?.name,
```
[the remark this answers is not in the document]
Comment (egor0798) [reply src/renderer/features/proxies/lib/proxies-utils.ts:42]: Checked it and reverted back

File: src/renderer/widgets/RemoveProxy/ui/Submit.tsx [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+        error: Boolean(params.multisigError),
+        callTimepoint: {
+          height: updatedTx.blockCreated || params.timepoint.height,
+          index: updatedTx.indexCreated || params.timepoint.index,
+        },
+      })
+      .catch(console.warn);
+  };
+
+  const sendProxynotification = () => {
```
Comment (Asmadek) [inline src/renderer/widgets/RemoveProxy/ui/Submit.tsx:139]: `sendProxyNotification`

**Review by Asmadek — APPROVED** [review]

**Review by tuul-wq — APPROVED** [review]
